### ISSUE TITLE
symlink network-wired to network-workgroup.svg to solve avahi issues

### ISSUE THREAD

**Issue by brittyazel** [numixproject/numix-icon-theme-circle#1963]
On Archlinux, Avahi is a base package and with it comes with three .desktop files, all which ask for network-wired for their icon. Unfortunately, this icon is horribly low quality in the default icon kits, as it has long since been deprecated. Even more unfortunate is that Avahi has not had a release in years, so has not corrected this issue.

When asking the Arch packagers to change the icon in the .desktop file, they said it would have to come from upstream, and I have gotten zero response from upstream.

A nice fix I have found, however, is to symlink a make-shift network-wired icon to the more popularly used network-workgroup.svg, as we then get a nicer icon that is scalable at least.

Could you please consider quickly adding this symlink in by default to help those of us on Archlinux with the Avahi conundrum?

**Comment by Foggalong** [comment]
@dirtydancing didn't we remove this symlink for some reason?

**Comment by dirtydancing** [comment]
@Foggalong We removed `nm-device-wireless.svg` from Numix-Circle, cf. https://github.com/numixproject/numix-icon-theme-circle/issues/1361, if that is what you were thinking of.

@brittyazel I can feel your visual pain :), e.g. this is a screenshot of Arch-based Antergos Live Session (ISO 2015.04.12) (VM) with GNOME Shell 3.16.0, all default (Avahi icons marked):

![antergos_avahi_default](https://cloud.githubusercontent.com/assets/7164227/7177937/2fd142f8-e42b-11e4-9db2-555c97bf4f65.png)

And this is probably what you are thinking of:

![antergos_avahi_changed](https://cloud.githubusercontent.com/assets/7164227/7177954/49cdcbc2-e42b-11e4-932d-0d1a94490c23.png)

What I did to get there is to change the icon line of all three .desktop files from `network-wired` to `network-workgroup`.

In Numix-Circle, `network-wired.svg` cannot simply be added, because `network-wired.svg` is an icon in Numix base (and I think that `network-wired.svg` is correctly located in Numix base, just like e.g. the corresponding `network-wireless.svg`), and as such is used e.g. in Ubuntu for the network settings, cf. e.g. https://github.com/numixproject/numix-icon-theme/issues/458.

I checked in daily Ubuntu 15.04 (VM): adding a symlink `network-wired.svg` with the target `network-workgroup.svg` in Numix-Circle will change the icon in the Ubuntu network settings to the Circle icon, which is the incorrect icon.

Current Numix-Circle (icon from Numix base):

![ubuntu1504_default](https://cloud.githubusercontent.com/assets/7164227/7178273/cc825d2e-e42d-11e4-987f-42e6d63c4263.png)

With added symlink `network-wired.svg` (incorrectly displaying a Circle icon):

![ubuntu1504_changed](https://cloud.githubusercontent.com/assets/7164227/7178292/e84ce15a-e42d-11e4-9eb8-e4059bcca71e.png)

This appears to be an upstream bug in Avahi indeed, and should be fixed by the Avahi devs. As long as this is not fixed upstream, a "workaround" is to change the icon line, just like I did above.

In conclusion: To me, this issue, while understandable, in Numix-Circle unfortunately is a "won't fix".

**Comment by dirtydancing** [comment]
Maybe another approach to address this would be to add `network-wired.svg` in larger sizes to Numix base, so that this icon scales better for Avahi in Arch, until fixed upstream. Currently, this looks like a smaller icon from Numix base is displayed, and apparently this does not scale well.

But that would still not be a Numix-Circle/Square icon, and moreover that would still be Numix "fixing" a bug that is not a Numix bug.

Anyways, if this approach in Numix base should be considered, such a new issue would need to be opened in Numix base, not in Numix-Circle.

**Comment by wa4557** [comment]
Or remove remove `network-wired` from status and add symlinks to `nm-device-wired.svg` in devices, as I proposed here :) https://github.com/numixproject/numix-icon-theme/issues/458

**Comment by dirtydancing** [comment]
@wa4557 Of course, I already linked to that issue above: that is still all taking place in Numix base :)

**Comment by palob** [comment]
@brittyazel I think the best workaround is to just hide the avahi launchers. You won't need them anyway, will you?
Edit the launchers *bssh.desktop*, *bvnc.desktop* and *avahi-discover.desktop* in */usr/share/applications*

Adding the line **NoDisplay=true** will prevent the launchers from appearing in a menu / the GNOME Shell overview.
You can do this as well with menu editors like Alacarte, Menu Libre, Arronax etc.

**Comment by brittyazel** [comment]
@palob Yes, I have done this numerous times in the past. I have hidden the launchers, changed the icon the .desktop points to, but each and every time the package updates to a new revision they get overwritten. 

I also agree that the best solution is for Avahi upstream to fix the problem, but, like I said before, the project hasn't had a release in quite a few years, and seems to be dormant. Nobody is there to respond to requests or make any changes whatsoever. And likewise they have their bug system locked down so that you can't even make an account to report a bug.

**Comment by palob** [comment]
In the Numix base theme a ``network-wired.svg`` is only present for 16x16, 22x22 and 24x24.
Could it be of any help to have one for other sizes or scalable as well in a way that the icon would show up at least sharp?

**Comment by dirtydancing** [comment]
@brittyazel Please note that, as explained above, currently `network-wired.svg` is an icon in Numix base, and having it as well in Numix-Circle would cause an icon conflict, because Numix-Circle comes before Numix base in the inheritance line, so `network-wired.svg` from Numix base would not be displayed anymore, because `network-wired.svg` from Numix-Circle would get picked up instead.

This means that within the current setup between Numix-Circle and Numix base, `network-wired.svg` cannot be added to Numix-Circle.

@palob Well, that is exactly what I suggested as a possible approach above :) (without specifying the exact sizes, though). Still, that would have to be dealt with in Numix base, not in Numix-Circle.

In addition, at the moment, the design for exactly that icon is in discussion in Numix base, cf. https://github.com/numixproject/numix-icon-theme/issues/458.

And indeed, having `network-wired.svg` in larger sizes in Numix base should be of help here, because then the icon should at least show up sharp (would probably need testing). In comparison, in Numix base `network-wireless.svg` is present up to `48x48`, and `network-wired.svg` could be added up to `256x256`, e.g. as a symlink to the already existing `nm-device-wired.svg` in `devices` (probably not in `scalable` though, as this is not a "symbolic"-named icon), details to be determined.

**Comment by dirtydancing** [comment]
One additional thought/idea: At the moment, in Numix base there is also an issue report against the current design of `nm-device-wired.svg` in `devices` (possible target for a symlink `network-wired.svg`), cf. https://github.com/numixproject/numix-icon-theme/issues/461.

In combination with this issue, this is an interesting timing, because `nm-device-wired.svg` in `devices` could be designed in a way that is also in harmony with the Circle/Square icons, so both of theses issues could be addressed at once. To my mind comes e.g. a "squarish" design of a network plug socket.

**Comment by dirtydancing** [comment]
Some progress is in the making: this pull request https://github.com/numixproject/numix-icon-theme/pull/462 turns `network-wired.svg` into symlinks with the target `nm-device-wired.svg`. I checked in a current Arch-based Antergos (VM) installation, and this is what the result looks like, using these two sources:

https://github.com/wa4557/numix-icon-theme/tree/bug458
https://github.com/numixproject/numix-icon-theme-circle

![antergos_avahi](https://cloud.githubusercontent.com/assets/7164227/7203294/ec85f6a2-e519-11e4-8307-11b62b25b7f5.png)

As you can see, the Avahi icons in the overview are at least sharp with that approach. This is already much better than in the initial screenshot above (current Numix base master). I still think that the icon `nm-device-wired.svg` itself could be improved, cf. https://github.com/numixproject/numix-icon-theme/issues/461.

**Comment by dirtydancing** [comment]
More progress: after https://github.com/numixproject/numix-icon-theme/issues/461 and https://github.com/numixproject/numix-icon-theme/pull/467, if merged, this is what it would look like:

![antergos_avahi_new](https://cloud.githubusercontent.com/assets/7164227/7220225/2ba011ea-e6c0-11e4-82a7-8ca48e885e71.png)

**Comment by dirtydancing** [comment]
@brittyazel Please note that while this issue has been closed in Numix-Circle, this is still dealt with in Numix base, cf. the related current state of affairs over at https://github.com/numixproject/numix-icon-theme/pull/468.

Here are two example screenshots of Arch-based Antergos from over there:

![ac36862c-e86e-11e4-9ee9-18a42c653429](https://cloud.githubusercontent.com/assets/7164227/7282133/2e99d014-e92e-11e4-9772-e44f26c7ca3f.png)

![e8a50870-e870-11e4-9e96-9c48315b0eee](https://cloud.githubusercontent.com/assets/7164227/7282142/3dc9256c-e92e-11e4-9e2b-acf303c6c0f8.png)

**Comment by Foggalong** [comment]
:+1:
